src/core/sound_recording_controller.py

```diff
@@ -0,0 +1,106 @@
+# coding: utf8
+
+__contact__ = "info@hytech-imaging.fr"
+__copyright__ = "Copyright (c) 2021 Hytech Imaging"
+
+import os.path
+from .thread_sound_recording import ThreadForSoundRecording
+from qgis.PyQt.QtCore import pyqtSignal, QObject
+from datetime import datetime
+
+
+class SammoSoundRecordingController(QObject):
+    onStopSoundRecordingForObservationSignal = pyqtSignal(str)
+
+    def __init__(self):
+        super().__init__()
+        self._workingDirectory: str = None
+        self._threadSoundRecording = self.createSoundRecording()
+        self._currentSoundFileName: str = None
+        self._startTimerOnRecordForCurrentObservation: float = None
+
+    def unload(self):
+        if self._threadSoundRecording.isProceeding:
+            self.onAutomaticStopRecordingTimerEnded()
+
+    def createSoundRecording(self) -> ThreadForSoundRecording:
+        threadSoundRecording = ThreadForSoundRecording(
+            self.onAutomaticStopRecordingTimerEnded
+        )
+        return threadSoundRecording
+
+    def onChangeObservationStatus(self, observationStatus: bool):
+        if observationStatus and not self._threadSoundRecording.isProceeding:
+            # on start observation when no sound recording is in progress
+            self.changeSoundRecordingStatus(True)
+
+        elif observationStatus and self._threadSoundRecording.isProceeding:
+            # on start observation when sound recording is in progress
+            self._threadSoundRecording.setAutomaticStopTimerSignal.emit(
+                -1
+            )  # cancel automatic stop
+            if self._startTimerOnRecordForCurrentObservation > 0:
+                soundRecordingDatas = (
+                    "{} - between the seconds {:.1f} and {:.1f} ".format(
```

Comment:
What do you think about adding a `sound_start` and `sound_end` fields instead of using string comments?

Reply:
I suppose you want to make the code more readable (and remove the comments), but I don't understand what is the use of fields.
Did you mean methods ?

Reply:
I was thinking about 2 new columns in the underlying database.